kubernetes/apps_kubeclient.go

```diff
@@ -770,63 +867,433 @@ func (kc *kubeClient) getPods(namespace string, uid types.UID) ([]v1.Pod, error)
 			}
 		}
 		if match {
-			appPods = append(appPods, pod)
+			appPods = append(appPods, &pod)
 		}
 	}
 
 	return appPods, nil
 }
 
-func (kc *kubeClient) getPodStatus(pods []v1.Pod) (*app.PodStats, error) {
-	var starting, running, stopping int
+// Pod status constants
+const (
+	podRunning     = "Running"
+	podNotReady    = "Not Ready"
+	podWarning     = "Warning"
+	podError       = "Error"
+	podPulling     = "Pulling"
+	podPending     = "Pending"
+	podSucceeded   = "Succeeded"
+	podTerminating = "Terminating"
+	podUnknown     = "Unknown"
+)
+
+func (kc *kubeClient) getPodStatus(pods []*v1.Pod) ([][]string, int) {
```

Comment:
is this method covered by 1 or more tests ?

Reply:
I have someone on my team writing tests, and I will get him to add coverage for this.
I would like to do that as a separate testing PR - this PR is focused on JSON-API compatibility.

Reply:
A whitebox test has not yet been written.